### PR TITLE
update membership rules

### PR DESCRIPTION
**Pull request by nikomatsakis**
Tyler and I were reviewing these membership criteria and we felt that the wording could be improved.

### CODE REVIEW

File: src/membership.md [hunk 1]
```diff
@@ -37,17 +37,19 @@ These are the questions we ask ourselves when deciding whether someone would be
...
         * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
-* Is this person able to **disagree productively**?
+* Is this person able to **disagree constructively and empathically**?
     * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
-    * Do they "steelman", looking for ways to restate others' points in the most convincing way? 
+    * Do they "steelman", looking for ways to restate others' points in the most convincing way?
+    * Do people come away from disagreements with this person feeling respected?
 * Is this person **active**?
-    * Are they attending the triage/design meetings regularly?
+    * Are they attending the [triage meeting](./meetings/triage.md) and [design meetings](./meetings/triage.md) regularly?
```
Comment (nikomatsakis) [inline src/membership.md:50]:
```suggestion
    * Are they attending the [triage meeting](./meetings/triage.md) and [design meetings](./meetings/triage.md) regularly? (meetings are open for anyone to attend; but note that merely attending meetings is not enough to become a team member!)
```

Comment (nikomatsakis) [reply src/membership.md:50]: Maybe add this?

**Comment by nikomatsakis** [comment]
Looks like our github action is messed up :(

File: src/membership.md [hunk 2]
```diff
@@ -37,17 +37,19 @@ These are the questions we ask ourselves when deciding whether someone would be
...
     * Either in meeting or elsewhere, do they comment on disussions and otherwise?
 * Does this person have an **overall desire to improve the language**, rather than a strong interest in some particular domain?
     * Everyone have preferences, but members are responsible for balancing a wide array of interests. Someone with very specialized interest may be a better choice for a lang team advisor.
+
+Keep in mind that qualifications are not a checklist and membership decisions are ultimately made on a case-by-case basis. **If you are interested in joining the lang team, we recommend you out to the lead(s) to talk about the path forward.**
```
Comment (RalfJung) [inline src/membership.md:55]:
```suggestion
Keep in mind that qualifications are not a checklist and membership decisions are ultimately made on a case-by-case basis. **If you are interested in joining the lang team, we recommend you reach out to the lead(s) to talk about the path forward.**
```

File: src/membership.md [hunk 3]
```diff
@@ -37,17 +37,19 @@ These are the questions we ask ourselves when deciding whether someone would be
...
     * Have they led an impactful initiative to completion?
-    * Have they identified a criticial flaw in someone else's design that led to the design being altered?
 * Is this person **responsible**?
     * When they agree to take on a task, do they either get it done or identify that they are not able to follow through and ask for help?
 * Is this person able to **lead others to a productive conversation**?
     * Are there times when a conversation was stalled out and this person was able to step in and get the design discussion back on track?
         * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
-* Is this person able to **disagree productively**?
+* Is this person able to **disagree constructively and empathically**?
     * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
```
Comment (nikomatsakis) [inline src/membership.md:46]:
```suggestion
    * The expectation is that team members go "above and beyond" the [Rust code of conduct](https://www.rust-lang.org/policies/code-of-conduct), embodying not only the letter but also the spirit.
    * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
```

Comment (nikomatsakis) [reply src/membership.md:46]: Maybe add that?

File: src/membership.md [hunk 4]
```diff
@@ -37,17 +37,19 @@ These are the questions we ask ourselves when deciding whether someone would be
...
-    * Do they "steelman", looking for ways to restate others' points in the most convincing way? 
+    * Do they "steelman", looking for ways to restate others' points in the most convincing way?
```
Comment (nikomatsakis) [inline src/membership.md:47]:
```suggestion
    * If they have a concern, do they engage actively to make sure it is understood and to look for ways to resolve it?
```
This feels repetitive, and I think this is an important point

File: src/membership.md [hunk 5]
```diff
@@ -37,17 +37,20 @@ These are the questions we ask ourselves when deciding whether someone would be
 
 * Has this person demonstrated **strong language design skills**?
     * Have they led an impactful initiative to completion?
-    * Have they identified a criticial flaw in someone else's design that led to the design being altered?
```
Comment (tmandry) [inline src/membership.md:40]: I don't know if I want to reduce this down to only the "leading to completion" part. What about something like

> Are they able to identify flaws in a design and, just as importantly, come up with creative solutions?

Though I'm not sure if that hits the mark entirely. "Creative" could be interpreted in a lot of ways. It's possible to come up with something creative that's also very complex, for example.

File: src/membership.md [hunk 6]
```diff
@@ -37,17 +37,20 @@ These are the questions we ask ourselves when deciding whether someone would be
...
-* Is this person able to **disagree productively**?
+* Is this person able to **disagree constructively and empathically**?
+    * The expectation is that team members go "above and beyond" the [Rust code of conduct](https://www.rust-lang.org/policies/code-of-conduct), embodying not only the letter but also the spirit.
     * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
-    * Do they "steelman", looking for ways to restate others' points in the most convincing way? 
+    * If they have a concern, do they engage actively to make sure it is understood and to look for ways to resolve it?
+    * Do people come away from disagreements with this person feeling respected?
 * Is this person **active**?
-    * Are they attending the triage/design meetings regularly?
+    * Are they attending the [triage meeting](./meetings/triage.md) and [design meetings](./meetings/triage.md) regularly? (meetings are open for anyone to attend; but note that merely attending meetings is not enough to become a team member!)
```
Comment (tmandry) [inline src/membership.md:51]:
```suggestion
    * Are they attending the [triage meeting](./meetings/triage.md) and [design meetings](./meetings/triage.md) regularly? (Meetings are open for anyone to attend, but note that merely attending meetings is not enough to become a team member!)
```

**Comment by scottmcm** [comment]
> Looks like our github action is messed up :(

Yeah, https://github.com/rust-lang/lang-team/issues/221

File: src/membership.md (hunk 5 above)
Comment (nikomatsakis) [reply src/membership.md:40]: Hmm. Yes, I like that. I'm also thinking that "leading to completion" isn't *necessarily* key -- I feel like it's ok to drive an important thing that winds up being abandoned, as long as that is actively decided, and the work in the interim feels good.

**Review by tmandry — APPROVED** [review]

**Comment by joshtriplett** [comment]
Reviewing right now; these changes largely seem like improvements, though there are a few bits being removed that I'd like to preserve.

File: src/membership.md [hunk 7]
```diff
@@ -36,18 +36,22 @@ Lang team members can propose new additions to the team as follows:
 These are the questions we ask ourselves when deciding whether someone would be a good choice as a lang team member.
 
 * Has this person demonstrated **strong language design skills**?
-    * Have they led an impactful initiative to completion?
-    * Have they identified a criticial flaw in someone else's design that led to the design being altered?
+    * Have they made major contributions to an area of the language, such as leading an impactful initiative to completion?
```
Comment (joshtriplett) [inline src/membership.md:39]:
```suggestion
    * Have they made notable contributions to an area of the language, such as leading an impactful initiative to completion?
```

Or perhaps some other word would work better; the previous text just said "led an impactful initiative", but calling it "major" feels like a further qualification. I think we'd value someone who had an impact in an area, even if that area itself wasn't "major". The important thing is a chance to see their design and shepherding skills in action contributing to the language.

File: src/membership.md [hunk 8]
```diff
@@ -36,18 +36,22 @@ Lang team members can propose new additions to the team as follows:
...
+    * Are they able to identify flaws in a design and, just as importantly, come up with creative solutions?
 * Is this person **responsible**?
     * When they agree to take on a task, do they either get it done or identify that they are not able to follow through and ask for help?
 * Is this person able to **lead others to a productive conversation**?
     * Are there times when a conversation was stalled out and this person was able to step in and get the design discussion back on track?
         * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
-* Is this person able to **disagree productively**?
+* Is this person able to **disagree constructively and empathically**?
```
Comment (joshtriplett) [inline src/membership.md:46]:
```suggestion
* Is this person able to **disagree constructively, with empathy**?
```

File: src/membership.md [hunk 9]
```diff
@@ -36,18 +36,22 @@ Lang team members can propose new additions to the team as follows:
...
+    * The expectation is that team members go "above and beyond" the [Rust code of conduct](https://www.rust-lang.org/policies/code-of-conduct), embodying not only the letter but also the spirit.
     * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
```
Comment (joshtriplett) [inline src/membership.md:48]:
```suggestion
    * Do they help turn disagreements into collaborations, jointly seeking a mutually satisfying solution based on everyone's values?
    * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
```

File: src/membership.md [hunk 10]
```diff
@@ -36,18 +36,22 @@ Lang team members can propose new additions to the team as follows:
...
-    * Do they "steelman", looking for ways to restate others' points in the most convincing way? 
+    * If they have a concern, do they engage actively to make sure it is understood and to look for ways to resolve it?
```
Comment (joshtriplett) [inline src/membership.md:49]:
```suggestion
    * If they *or others* have a concern, do they engage actively to make sure it is understood and to look for ways to resolve it?
```

File: src/membership.md [hunk 11]
```diff
@@ -36,18 +36,22 @@ Lang team members can propose new additions to the team as follows:
...
+    * Do people come away from disagreements with this person feeling respected?
```
Comment (joshtriplett) [inline src/membership.md:50]:
```suggestion
    * Do people come away from disagreements with this person feeling respected, understood, and valued?
```

File: src/membership.md [hunk 12]
```diff
@@ -36,18 +36,22 @@ Lang team members can propose new additions to the team as follows:
...
 * Is this person **responsible**?
     * When they agree to take on a task, do they either get it done or identify that they are not able to follow through and ask for help?
 * Is this person able to **lead others to a productive conversation**?
     * Are there times when a conversation was stalled out and this person was able to step in and get the design discussion back on track?
         * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
-* Is this person able to **disagree productively**?
+* Is this person able to **disagree constructively and empathically**?
+    * The expectation is that team members go "above and beyond" the [Rust code of conduct](https://www.rust-lang.org/policies/code-of-conduct), embodying not only the letter but also the spirit.
     * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
-    * Do they "steelman", looking for ways to restate others' points in the most convincing way?
```
Comment (joshtriplett) [inline src/membership.md:48]: At first it bothered me to lose "steelman" here, but I think with the addition I made to the next point ("if they *or others* have a concern ..."), together with the addition above about collaborative disagreement, I think it's probably sufficiently captured without the jargon.

File: src/membership.md (hunk 8 above)
Comment (nikomatsakis) [reply src/membership.md:46]: I like the word collaboratively! Nice.

Comment (nikomatsakis) [reply src/membership.md:46]: I wonder if we should also include the language "treasure dissent"

File: src/membership.md (hunk 12 above)
Comment (nikomatsakis) [reply src/membership.md:48]: Sounds good.

File: src/membership.md [hunk 13]
```diff
@@ -36,18 +36,23 @@ Lang team members can propose new additions to the team as follows:
...
         * This could have been by suggesting a compromise, but it may also be by asking the right questions or encouraging the right tone.
-* Is this person able to **disagree productively**?
+* Is this person able to **disagree collaboratively, constructively, and with empathy**?
+    * The expectation is that team members go "above and beyond" the [Rust code of conduct](https://www.rust-lang.org/policies/code-of-conduct), embodying not only the letter but also the spirit.
+    * Do they help turn disagreements into collaborations, jointly seeking a mutually satisfying solution based on everyone's values?
     * When they are having a debate, do they make an active effort to understand and repeat back others' points of view?
-    * Do they "steelman", looking for ways to restate others' points in the most convincing way? 
+    * If they *or others* have a concern, do they engage actively to make sure it is understood and to look for ways to resolve it?
+    * Do people come away from disagreements with this person feeling respected, understood, and valued?
```
Comment (traviscross) [inline src/membership.md:51]:
```suggestion
    * Do they respect others when disagreeing, seek earnestly to understand others' points of view, and show that they value others for bringing forward reasonable disagreement and dissent?
```

It's generally healthier to phrase things in terms of what people can control.  E.g., it's always possible to respect others when disagreeing, but it's not always possible to control how others feel about that disagreement.

Comment (nikomatsakis) [reply src/membership.md:51]: Thank you. I agree, and the initial phrasing was bothering me in this regard, but I didn't know how to do better.

File: src/membership.md (hunk 8 above)
Comment (joshtriplett) [reply src/membership.md]: @nikomatsakis I would love to see "treasure dissent" included if you can find a good place for it.

**Comment by nikomatsakis** [comment]
Going to take a chance and merge.

**Pull request merged**